skills_DS/base/views.py

```diff
@@ -2,17 +2,68 @@
 from . import forms
 from django.contrib import messages
 
+from django.views import View
+from django.contrib.auth import get_user_model
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from django.contrib.sites.shortcuts import get_current_site
+from .utils import token_generator
+from django.core.mail import send_mail
+from django.template.loader import render_to_string
+from django.utils.encoding import force_bytes, force_str
+from django.http import HttpResponse
+import hashlib
+from django.core.mail import EmailMessage
+from django.urls import reverse
+
 # Create your views here.
 
+
 def register(request):
+    # User = get_user_model()
     if request.method == 'POST':
         form = forms.UserRegisterForm(request.POST)
         if form.is_valid():
-            form.save()
+            user = form.save(commit=False)
+            user.is_active = False
+            user.save()
+
+            current_site = get_current_site(request)
+            mail_subject = 'Activate your account.'
+
+            message = render_to_string('base/email_template.html', {
+                'user': user,
+                'domain': current_site.domain,
+                'uid': urlsafe_base64_encode(force_bytes(user.pk)),
+                'token': token_generator.make_token(user),
+            })
+            to_email = form.cleaned_data.get('email')
+            send_mail(mail_subject, message, 'cosc448atesting@gmail.com', [to_email])
```

Comment:
We should probably store the email address somewhere else other than hard coding it.

Reply:
There's a setting option we are already using, so we should use that instead.